tree/30_generic_methods/file_report_content.cf

```diff
@@ -0,0 +1,106 @@
+#####################################################################################
+# Copyright 2019 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name File report content
+# @description Report the content of a file
+# @documentation Reports the content of a file.
+#
+# This method does nothing on the system, but only reports a complete or partial content
+# from a given file. This allows centralizing this information on the searver, and avoid
+# having to connect on each node to get this information.
+# 
+# #### Parameters
+#
+# ##### Target
+#
+# This is the file you want to report content from. The method will return an error if it
+# does not exist.
+#
+# ##### Regex
+#
+# If empty, the method will report the whole file content.
+# If set, the method will grep the file for the given regular expression, and
+# report the result.
+#
+# ##### Context
+#
+# When specifying a regex, will add the number of lines of context around matches
+# (default is 0, i.e. no context).
+#
+# When reporting the whole file, this parameter is ignored.
+#
+# #### Examples
+# 
+# ```
+# # To get the whole /etc/hosts content
+# file_report_content("/etc/hosts", "", "");
+# # To get lines starting by "nameserver" in /etc/resolv.conf
+# file_report_content("/etc/resolv.conf", "^nameserver", "");
+# # To get lines containing "rudder" from /etc/hosts with 3 lines of context
+# file_report_content("/etc/hosts", "rudder", "3");
+# ```
+#
+# @parameter target     File to report content from
+# @parameter regex      Regex to search in the file (empty for whole file)
+# @parameter_constraint regex "allow_empty_string" : true
+# @parameter context    Number of context lines when matching regex (default is 0)
+# @parameter_constraint context "allow_empty_string" : true
+# @parameter_constraint context "regex" : "^\\d*$"
+#
+# @class_prefix file_report_content
+# @class_parameter target
+
+bundle agent file_report_content(target, regex, context)
+{
+  vars:
+      "old_class_prefix"  string => canonify("file_report_content_${target}");
+      "args"               slist => { "${target}", "${regex}", "${context}" };
```

Comment:
this should be in !pass1 is we don't want to rewrite the class prefix based on default (like in directory_absent)